flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java

```diff
@@ -310,6 +311,27 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
                                             field.name(),
                                             field.type(),
                                             update.newDescription()));
+                } else if (change instanceof UpdateColumnPosition) {
+                    UpdateColumnPosition update = (UpdateColumnPosition) change;
+                    SchemaChange.Move move = update.move();
+
+                    // key: name ; value : index
+                    Map<String, Integer> map = new HashMap<>();
+                    for (int i = 0; i < newFields.size(); i++) {
+                        map.put(newFields.get(i).name(), i);
+                    }
+
+                    int fieldIndex = map.get(move.fieldName());
+                    int refIndex = 0;
+                    if (move.type().equals(SchemaChange.Move.MoveType.FIRST)) {
+                        checkMoveIndexEqual(move, fieldIndex, refIndex);
+                        newFields.add(refIndex, newFields.remove(fieldIndex));
+                    } else if (move.type().equals(SchemaChange.Move.MoveType.AFTER)) {
+                        refIndex = map.get(move.referenceFieldName());
+                        checkMoveIndexEqual(move, fieldIndex, refIndex);
+                        newFields.add(refIndex + 1, newFields.remove(fieldIndex));
```

Comment:
Maybe bug here?
Total has 4 fields.
fieldIndex = 1;
refIndex = 2;
f0, f1, f2, f3
after remove: f0 f2 f3
and add(3)? becoming f0 f2 f3 f1?